R/deploy.R

```diff
@@ -487,49 +487,65 @@ has_image <- function(content) {
 #'
 #' Set the Content Image using a variety of methods.
 #'
-#' NOTE: `set_image_webshot()` requires [webshot2::webshot()], but currently
-#' skips and warns for any content that requires authentication until the
-#' `webshot2` package supports authentication.
-#'
 #' @param content A content object
-#' @param path The path to an image on disk
-#' @param url The url for an image
-#' @param ... Additional arguments passed on to [webshot2::webshot()]
+#' @param path A file path or URL to an image
 #'
-#' @rdname set_image
 #' @family content functions
+#' @rdname set_content_image
 #' @export
-set_image_path <- function(content, path) {
-  warn_experimental("set_image_path")
+set_content_image <- function(content, path) {
+  warn_experimental("set_content_image")
   validate_R6_class(content, "Content")
-  guid <- content$get_content()$guid
 
+  valid_path <- NULL
+  if (file.exists(path)) {
+    valid_path <- path
+  } else {
+    parsed <- httr::parse_url(path)
+    if (parsed$scheme %in% c("http", "https")) {
+      valid_path <- fs::file_temp(pattern = "image", ext = fs::path_ext(parsed[["path"]]))
+      httr::GET(url, httr::write_disk(valid_path))
+      on.exit(unlink(valid_path))
+    }
+  }
+  if (is.null(valid_path)) {
+    stop("Could not locate image at `path`")
```

Comment:
Would it be helpful to put the actual path in this message too?

Reply:
Of course 😬